include/tvm/ir/type_relation.h

```diff
@@ -203,7 +203,6 @@ class TypeRelationNode : public TypeConstraintNode {
 
   bool SEqualReduce(const TypeRelationNode* other, SEqualReducer equal) const {
     return
-        equal(func, other->func) &&
```

Comment:
Let us keep this comparison. we can fix the EnvFunc's SEqual instead by comparing the name.